node/hyperbahn/service_proxy.js

```diff
@@ -424,6 +433,63 @@ function isExitFor(serviceName) {
     return chan.serviceProxyMode === 'exit';
 };
 
+ServiceDispatchHandler.prototype.handleCircuitStateChange =
+function handleCircuitStateChange(change) {
+    var self = this;
+
+    var circuit = change.circuit;
+    var oldState = change.oldState;
+    var state = change.state;
+
+    if (oldState && oldState.healthy !== state.healthy) {
+        // unhealthy -> healthy
+        if (state.healthy) {
+            self.statsd.increment('circuits.healthy.total', 1);
```

Comment:
Please use StatEmitter instead of statsd directly. Once m3 is ready we may move away from statsd which has a difference interface compared to m3.